flaml/nlp/autotransformers.py

```diff
@@ -0,0 +1,839 @@
+
+        self._save_ckpt_json(best_ckpt)
+
+        if verbose==0:
+            logger.setLevel(old_level)
+
+        return validation_metric, analysis
+
+    def predict(self,
+                ckpt_json_dir = None,
+                **kwargs):
+        '''Predict label for test data.
+
+        An example:
+            predictions, test_metric = autohf.predict()
+
+        Args:
+            ckpt_json_dir:
+                the checkpoint for the fine-tuned huggingface if you wish to override the saved checkpoint in the training stage under self.path_utils._result_dir_per_run
+
+        Returns:
+            A numpy array of shape n * 1 - - each element is a predicted class
+            label for an instance.
+        '''
+        best_checkpoint = self._load_ckpt_json(ckpt_json_dir, **kwargs)
+        best_model = self._load_model(checkpoint_path=best_checkpoint)
+        training_args = TrainingArguments(per_device_eval_batch_size=1,
+                                          output_dir=self.path_utils.result_dir_per_run)
+        test_trainer = TrainerForAutoTransformers(best_model, training_args)
+
+        if self.jobid_config.spt == "ori":
+            try:
+                self.test_dataset.remove_columns_("label")
+            except ValueError:
+                pass
+
+        test_dataloader = test_trainer.get_test_dataloader(self.test_dataset)
+        predictions, labels, _ = test_trainer.prediction_loop(test_dataloader, description="Prediction")
+        predictions = np.squeeze(predictions) \
+            if get_default_task(self.jobid_config.dat[0], self.jobid_config.subdat) == "regression" \
+            else np.argmax(predictions, axis=1)
+        torch.cuda.empty_cache()
+
+        if self.jobid_config.spt == "rspt":
+            assert labels is not None
+            metric = self._get_metric_func()
+            output_metric = metric.compute(predictions=predictions, references=labels)
+            self._save_output_metric(output_metric)
+            return predictions, output_metric
+        else:
+            return predictions, None
+
+    def output_prediction(self,
+                          predictions,
+                          output_prediction_path,
+                          output_dir_name):
+        """
+            When using the original GLUE split, prediction and prepare the submission file
+
+            Example:
+                local_archive_path = self.autohf.output_prediction(predictions,
+                                      output_prediction_path= self.console_args.data_root_dir + "result/",
+                                      output_dir_name=azure_save_file_name)
+
+            Args:
+                predictions:
+                    a list of predictions, which is the output of AutoTransformers.predict()
+                output_prediction_path:
+                    output path for the prediction
+                output_dir_name:
```

Comment:
diff between output_prediction_path and output_dir_name is not obvious from the name